### PR TITLE
Repro for #15118: Incorrect result of divide or multiply two CumulativeSum expressions

### PR DESCRIPTION
**Pull request by camsaul**
Backend reproduction for #15118

This was done as part of a live demo for @nemanjaglumac on writing backend tests. Keep your eyes peeled for the video. Is it too soon to whisper Oscar?

Edit:
### Note:
- Fix the test with the real values when the issue is fixed
- Uncomment the test and merge it together with the fix

### CODE REVIEW

File: test/metabase/query_processor_test/expression_aggregations_test.clj [hunk 1]
```diff
@@ -269,8 +269,36 @@
   (mt/test-drivers (mt/normal-drivers-with-feature :expression-aggregations)
     (testing "can we use named expressions inside expression aggregations?"
       (is (= [[406]]
-             (mt/format-rows-by [int]
-               (mt/rows
-                 (mt/run-mbql-query venues
-                   {:aggregation [[:sum [:expression "double-price"]]]
-                    :expressions {"double-price" [:* $price 2]}}))))))))
+             (mt/formatted-rows [int]
```
Comment (nemanjaglumac) [inline test/metabase/query_processor_test/expression_aggregations_test.clj:272]: Does `mt/formatted-rows` automatically "extract" only rows from a query? Is that the reason we removed `mt/rows` in the previous step?

File: test/metabase/query_processor_test/expression_aggregations_test.clj [hunk 2]
```diff
@@ -269,8 +269,36 @@
...
+#_(deftest multiple-cumulative-sums-test
+  ;; sample-dataset doesn't work on Redshift yet -- see #14784
+  (mt/test-drivers (disj (mt/normal-drivers-with-feature :expression-aggregations) :redshift)
+    (testing "The results of divide or multiply two CumulativeSum should be correct (#15118)"
+      (mt/dataset sample-dataset
+        (is (= [["2016-01-01T00:00:00Z" 3236  2458.0  5694.0   1]
+                ["2017-01-01T00:00:00Z" 17587 14995.0 32582.0  2]
+                ["2018-01-01T00:00:00Z" 40381 35366.5 75747.5  3]
+                ["2019-01-01T00:00:00Z" 65835 58002.7 123837.7 4]
+                ["2020-01-01T00:00:00Z" 69540 64923.0 134463.0 5]]
```
Comment (nemanjaglumac) [inline test/metabase/query_processor_test/expression_aggregations_test.clj:286]: Maybe add back a comment that 1,2,3,4,5 are just placeholder values?

**Review by nemanjaglumac — APPROVED** [review]
Left a few comments but overall looks good to me :)

File: test/metabase/query_processor_test/expression_aggregations_test.clj (hunk 1 above)
Comment (camsaul) [reply test/metabase/query_processor_test/expression_aggregations_test.clj:272]: `mt/formatted-rows` is the same as `mt/rows` + `mt/format-rows-by` so I just consolidated them

**Pull request merged**